RegistrySubscriptionManager/src/main/java/eu/neclab/ngsildbroker/registry/subscriptionmanager/repository/RegistrySubscriptionInfoDAO.java

```diff
@@ -1,26 +1,301 @@
+								.sendJsonObject(new JsonObject(request.getContext().serialize()))
+								.onItemOrFailure().transformToUni((item, failure) -> {
+									if (failure != null) throw new RuntimeException();
+									String contextId = item.bodyAsString();
+									return client.preparedQuery(
+													"UPDATE registry_subscriptions SET subscription=subscription || $2, context=$3 WHERE subscription_id=$1 RETURNING subscription")
+											.execute(Tuple.of(request.getId(), new JsonObject(request.getPayload()),
+													contextId)).onItem().transform(i-> Tuple2.of(request.getPayload(),request.getContext().serialize().get("@context")));
+								}));
+	}
+
+	public Uni<RowSet<Row>> deleteSubscription(DeleteSubscriptionRequest request) {
+		return clientManager.getClient(request.getTenant(), false).onItem().transformToUni(client -> {
+			return client.preparedQuery("DELETE FROM registry_subscriptions WHERE subscription_id=$1")
+					.execute(Tuple.of(request.getId()));
+		});
+	}
+
+	public Uni<RowSet<Row>> getAllSubscriptions(String tenant, int limit, int offset) {
+		return clientManager.getClient(tenant, false).onItem().transformToUni(client -> {
+			return client.preparedQuery("SELECT subscription  FROM registry_subscriptions LIMIT $1 OFFSET $2")
+					.execute(Tuple.of(limit, offset)).onFailure().retry().atMost(3);
+		});
+	}
+
+	public Uni<RowSet<Row>> getSubscription(String tenant, String subscriptionId) {
+		return clientManager.getClient(tenant, false).onItem().transformToUni(client -> {
+			return client.preparedQuery("SELECT subscription FROM registry_subscriptions WHERE subscription_id=$1")
+					.execute(Tuple.of(subscriptionId)).onFailure().retry().atMost(3);
+		});
+	}
+
+	public static void main(String[] args) throws JsonGenerationException, IOException {
+		JsonLdProcessor.init("https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context-v1.6.jsonld");
+
+		Context context = JsonLdProcessor.getCoreContextClone();
+		Context test = new Context().parse(context.serialize().get("@context"), false);
+
+		System.out.println(JsonUtils.toPrettyString(context.serialize()));
+		System.out.println(JsonUtils.toPrettyString(test.serialize()));
 	}
 
-	@Override
-	protected StorageFunctionsInterface getStorageFunctions() {
-		return new RegistryStorageFunctions();
+	public Uni<RowSet<Row>> getRegById(String tenant, String id) {
+		return clientManager.getClient(tenant, false).onItem().transformToUni(client -> {
+			return client.preparedQuery("SELECT reg FROM csource WHERE id = $1").execute(Tuple.of(id)).onFailure()
+					.retry().atMost(3);
+		});
 	}
 
-	@Override
-	protected String getDBName() {
-		return "registry_subscriptions";
+	public Uni<Void> updateNotificationSuccess(String tenant, String id, String date) {
+		return clientManager.getClient(tenant, false).onItem().transformToUni(client -> {
+			return client
+					.preparedQuery("UPDATE registry_subscription SET subscription = subscription || ('{\""
```

Comment:
It should be "registry_subscriptions" instead of "registry_subscription".

Reply:
done